ap.md

```diff
@@ -0,0 +1,236 @@
+  var ledRegex = /leds/; // a Regex to catch requests to toggle LEDs
+
+  if (urlParts.pathname.match(ledRegex)) {
+    // if there is a request containing 'leds' call a function, toggleLED
+    toggleLED(urlParts.pathname, request, response);
+  } else {
+    // all other request will call a function, showIndex
+    showIndex(urlParts.pathname, request, response);
+  }
+});
+
+// Stays the same
+server.listen(8080);
+
+// Stays the same
+console.log('Server running at http://192.168.1.101:8080/');
+{% endhighlight %}
+
+Inside our `createServer` callback function, we've replaced the old plain text response with a basic router that catches requests to '/leds/:index' and defaults all other requests to a separate function. Let's add those functions below the `console.log`:
+
+{% highlight javascript %}
+// respond to the request with our index.html page
+function showIndex (url, request, response) {
+  response.writeHead(200, {"Content-Type": "text/html"});
+  // use fs to read in index.html
+  fs.readFile(__dirname + '/index.html', function (err, content) {
+    if (err) {
+      throw err;
+    }
+
+    // serves the content of index.html read in by fs
+    response.end(content);
+  });
+}
+
+// toggle the led specified in the url and respond with its state
+function toggleLED (url, request, response) {
+  var indexRegex = /(\d)$/; // Regex to find the number at the end of the url
+  var result = indexRegex.exec(url); // capture the number, returns an array
+  var index = +result[1]; // grab the captured result from the array and make sure it's a Number
+
+  var led = tessel.led[index]; // use the index to refence the correct LED
+
+  // toggle the state of the led and call the callback after that's done
+  led.toggle(function (err) {
+    if (err) {
+      // log the error and send back a 500 (internal server error) response to the client
+      console.log(err);
+      response.writeHead(500, {"Content-Type": "application/json"});
+      response.end(JSON.stringify({error: err}));
+    } else {
+      // the led was successfully toggled, so respond with the state of the toggled led using led.isOn
+      response.writeHead(200, {"Content-Type": "application/json"});
+      response.end(JSON.stringify({on: led.isOn}));
+    }
+  });
+}
+{% endhighlight %}
+
+In our `showIndex` function, we grab the content of `index.html` using the Node.js `fs` module and respond to the web browser request with that content. In our toggleLED function, we use a Regex to search the url for the index of the led we want to toggle and pass that index into the `tessel.led` array. Now that we have the led we want, we toggle it and use the callback to respond to the server request with the current state of the led using `isOn` property.
+
+One last step is to create a `.tesselignore` file in the project directory and add any non-JavaScript files we want to deploy with `ap.js`. For this project, that means adding one line: `index.html`. 
```

Comment:
I guess I don't know how .tesselignore works, because this seems backwards to me. We add the file we *want* to deploy by putting it in the ignore file?